src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -928,71 +978,19 @@ def get_create_and_insert_plan(child: SnowflakePlan, replace=False, error=True):
                     propagate_referenced_ctes=False,
                 )
             else:
-                return self.build(
-                    lambda x: create_table_as_select_statement(
-                        full_table_name,
-                        x,
-                        column_definition,
-                        replace=True,
-                        table_type=table_type,
-                        clustering_key=clustering_keys,
-                        comment=comment,
-                    ),
-                    child,
-                    source_plan,
-                    is_ddl_on_temp_object=is_temp_table_type,
-                    propagate_referenced_ctes=False,
-                )
+                return get_create_table_as_select_plan(child, replace=True, error=True)
+
         elif mode == SaveMode.OVERWRITE:
-            return self.build(
-                lambda x: create_table_as_select_statement(
-                    full_table_name,
-                    x,
-                    column_definition,
-                    replace=True,
-                    table_type=table_type,
-                    clustering_key=clustering_keys,
-                    comment=comment,
-                ),
-                child,
-                source_plan,
-                is_ddl_on_temp_object=is_temp_table_type,
-                propagate_referenced_ctes=False,
-            )
+            return get_create_table_as_select_plan(child, replace=True, error=True)
+
         elif mode == SaveMode.IGNORE:
-            return self.build(
-                lambda x: create_table_as_select_statement(
-                    full_table_name,
-                    x,
-                    column_definition,
-                    error=False,
-                    table_type=table_type,
-                    clustering_key=clustering_keys,
-                    comment=comment,
-                ),
-                child,
-                source_plan,
-                is_ddl_on_temp_object=is_temp_table_type,
-                propagate_referenced_ctes=False,
-            )
+            return get_create_table_as_select_plan(child, replace=False, error=False)
+
         elif mode == SaveMode.ERROR_IF_EXISTS:
-            if is_generated:
+            if creation_source == TableCreationSource.CACHE_RESULT:
```

Comment:
let's add a comment here about why we use the create and insert plan for cache result, which for handling cache_result during transaction, for the other internally generated table by large query break down, since the optimization will be blocked for query with active transaction, we can safely create it with select statement